modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/resource/BaseHeadlessBuilderResourceImpl.java

```diff
@@ -29,4 +34,10 @@ public abstract class BaseHeadlessBuilderResourceImpl {
 	@Produces({"application/json", "application/xml"})
 	public abstract Response get() throws Exception;
 
+	@Context
+	protected APIApplication contextAPIApplication;
+
+	@Context
```

Comment:
Not sure if in this case, the `@Context` annotation is necessary as AFAIK (as the Headless Builder Application is registered with `"osgi.jaxrs.extension.select", "(osgi.jaxrs.name=Liferay.Vulcan)"`)  the HttpServletRequest is being injected by https://github.com/liferay/liferay-portal/blob/c85ce289413642b25ff065256d94e14d9aebe932/modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/container/request/filter/ContextContainerRequestFilter.java#L338

Reply:
As we talked internally, the class that you mention should inject the context but for some reason it is not doing it because it is finding an instance. The contextContainerRequestFilter will inject the context in case there is no instance.

As we are a little in a hurry with this, I prefer to go forward with this, leaving the `@Context` annotation, and, in case we need or should remove the annotation, we can do it.

About your concern of injecting the dependency twice, it is not happening for the reason described above